internal/mode/static/nginx/config/servers.go

```diff
@@ -121,14 +152,15 @@ func createLocations(pathRules []dataplane.PathRule, listenerPort int32) []http.
 				matches = append(matches, match)
 			}
 
-			buildLocations = updateLocationsForFilters(r.Filters, buildLocations, r, listenerPort, rule.Path)
+			buildLocations = updateLocationsForFilters(r.Filters, buildLocations, r, server.Port, rule.Path)
 			locs = append(locs, buildLocations...)
 		}
 
 		if len(matches) > 0 {
-			matchesStr := convertMatchesToString(matches)
 			for i := range extLocations {
-				extLocations[i].HTTPMatchVar = matchesStr
+				key := server.Hostname + extLocations[i].Path + strconv.Itoa(int(server.Port))
```

Comment:
Using numbers would make it difficult to debug which matches belong to which server/location. Maybe there's a way we can still include some context info, along with a number to make it unique?

Reply:
would it be a big problem though? You can still quickly get which location the key belongs by searching for the key in the generated NGINX config.
Otherwise, we can include a metadata field in the match that can describe server hostname, port and location?  But it'd be great to see a use case users would use it.

Reply:
You're right, the key would still be in the nginx config. I think that's fine. I don't want to add too much info to it. I just had it my mind that if someone wanted to debug and look directly at the `match.json` file, they could know exactly what the match applies to. But cross-reference should still work.

Reply:
`serverId` meaning the server port ? Doesn't look like we have a ID for [server](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/internal/mode/static/state/dataplane/types.go#L60)

Reply:
@salonichf5 you can use the position of every server in the corresponding slice as its ID when we iterate over them here 
https://github.com/nginxinc/nginx-gateway-fabric/blob/b16d6aa5d24a6fd5a56cb0998f3eb8415a3cb5d2/internal/mode/static/nginx/config/servers.go#L50-L56

this will be similar to getting ID of a rule here https://github.com/nginxinc/nginx-gateway-fabric/blob/b16d6aa5d24a6fd5a56cb0998f3eb8415a3cb5d2/internal/mode/static/nginx/config/servers.go#L107

Reply:
I don't think it would be a proper fix for key cause we are storing all the keys and values in one match.json file across all servers, locations. For example :

Server 1 has multiple match locations the keys would be -- `1-1 , 1-2, 1-3` 
SSL Server 1 has single match location, the keys would be -- `1-1` 

The map will not store duplicate keys, are you okay if I add a `SSL` prefix for SSL servers to differentiate between these cases?

Reply:
> are you okay if I add a SSL prefix for SSL servers to differentiate between these cases?
yep, that makes sense

Reply:
@pleshakov I had to add `EXACT` to the `key` due to conflicts when exact path and prefix path resulted in same type.

Reply:
> @pleshakov I had to add EXACT to the key due to conflicts when exact path and prefix path resulted in same type.

in those cases, routing rules are the same, so we should be ok with having only one key. Unless I'm missing something.

Reply:
okay yeah that makes sense